test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ClientTests/Tests/ClientQueryTests.cs

```diff
@@ -0,0 +1,228 @@
+            {
+                services.ConfigureControllers(typeof(ClientQueryTestsController), typeof(MetadataController));
+
+                services.AddControllers().AddOData(opt => opt.EnableQueryFeatures()
+                    .AddRouteComponents("odata", CommonEndToEndEdmModel.GetEdmModel()));
+            }
+        }
+
+        public ClientQueryTests(TestWebApplicationFactory<TestsStartup> fixture)
+            : base(fixture)
+        {
+            _baseUri = new Uri(Client.BaseAddress, "odata/");
+            _context = new Container(_baseUri);
+            _context.HttpClientFactory = HttpClientFactory;
+        }
+
+        [Fact]
+        public void DollarFilter_UsingContains_ExecutesSuccessfully()
+        {
+            var testlists = new List<string>()
+            {
+                new("People?$filter=contains(Name, 'm')"),
+                new("People?$filter=contains('m', Name)"),
+                new("People?$filter=not contains(Name, 'm')"),
+                new("People?$filter=contains(Name, 'm') eq true"),
+                new("People?$filter=false eq contains(Name, 'm')"),
+                new("People?$filter=contains(Name, substring('name',  2, 1))"),
+                new("People?$filter=contains(concat('User','name'), 'name')"),
+                new("People?$filter=contains(concat('User','name'), substring('name',  2, 1))"),
+            };
+
+            foreach (var test in testlists)
+            {
+                var result = _context.Execute<Common.Clients.EndToEnd.Person>(new Uri(_baseUri.OriginalString + test)).ToArray();
+                Assert.NotNull(result);
+            }
+        }
+
+        [Fact]
+        public void Using_LinqContains_ExecutesSuccessfully()
+        {
+            var result = (from c in _context.People
+                          where c.Name.Contains("m")
+                          select new Common.Clients.EndToEnd.Person() { Name = c.Name }) as DataServiceQuery<Common.Clients.EndToEnd.Person>;
+
+            Assert.NotNull(result);
+
+            result = (DataServiceQuery<Common.Clients.EndToEnd.Person>)_context.People.Where(c => c.Name.Contains("m"));
+            Assert.NotNull(result);
+        }
+
+        [Fact]
+        public void Using_PrimitiveTypeInRequestUrl_ExecutesSuccessfully()
+        {
+            const string stringOfCast = "cast(PersonId,Edm.Byte)";
+
+            //GET http://jinfutanodata01:9090/AstoriaDefault635157546921762475/Person()?$filter=cast(cast(PersonId,Edm.Byte),Edm.Int32)%20gt%200 HTTP/1.1
+            //all the IDs in [-10, 2] except 0 are counted in.
+            var result = _context.People.Where(c => (Byte)c.PersonId > 0);
+            var stringOfQuery = result.ToString();
+            Assert.Contains(stringOfCast, stringOfQuery);
+            Assert.Equal(12, result.ToList().Count());
+
+            //GET http://jinfutanodata01:9090/AstoriaDefault635157551526070289/Person()?$filter=PersonId%20le%20256 HTTP/1.1

```

Comment:
I think we should also assert that this URL is generated instead of only having it as a comment.